### PR TITLE
[MP][OPS-6263] Platform catalogue upgrade

### PR DESCRIPTION
**Pull request by PucilowskiHMRC**

### CODE REVIEW

File: app/ssttpcalculator/CalculatorController.scala [hunk 1]
```diff
@@ -93,7 +93,11 @@ class CalculatorController @Inject() (
 
             journeyService.saveJourney(newJourney).map(
               _ => Redirect(ssttpcalculator.routes.CalculatorController.getMonthlyPayment()))
-
+          case PayTodayQuestion(None) =>
```
Comment (cavster) [inline app/ssttpcalculator/CalculatorController.scala:96]: This was needed ?

Comment (cavster) [reply app/ssttpcalculator/CalculatorController.scala:96]: We could make PayTodayQuestion not be an option if that's not too much work

File: app/ssttpcalculator/CalculatorService.scala [hunk 2]
```diff
@@ -230,8 +231,8 @@ class CalculatorService @Inject() (
     val sortedDebits: Seq[DebitInput] = debits.sortBy(_.dueDate)
 
       def processDebits(amount: BigDecimal, debits: Seq[DebitInput]): BigDecimal = {
-        debits match {
-          case debit :: Nil => calculateAmount(amount, debit)._1 * calculateDays(debit) * currentDailyRate
+        debits.toList match {
```
Comment (cavster) [inline app/ssttpcalculator/CalculatorService.scala:235]: Do we need recursion for this might be a easier way to write it...Also good pratice to add that @tail.annocate thing to make sure it  checks it will not blow up

File: app/ssttpcalculator/CalculatorController.scala (hunk 1 above)
Comment (PucilowskiHMRC) [reply app/ssttpcalculator/CalculatorController.scala:96]: upgrading scala version introduced a match error (inexhaustive match)

i'm unsure what case the optional boolean is supposed to encode when it is none

the larger issue at hand is older code abusing optionals (we've mostly dealt with this in dd-frontend now) and ssttp is yet to get this kind of love

File: app/testonly/TestUsersController.scala [hunk 3]
```diff
@@ -59,7 +59,7 @@ final case class TestUserForm(
     isOnIA                       = isOnIA,
     authorityId                  = authorityId.map(AuthorityId.apply).getOrElse(AuthorityId.random),
     affinityGroup                = AffinityGroup(affinityGroup),
-    confidenceLevel              = if (hasOverTwoHundred) L200.level else L100.level,
+    confidenceLevel              = if (hasOverTwoHundred) L200.level else L50.level,
```
Comment (cavster) [inline app/testonly/TestUsersController.scala:62]: Why was this changed to L50?

File: build.sbt [hunk 4]
```diff
@@ -54,7 +47,7 @@ lazy val microservice = Project(appName, file("."))
     ))
   .settings(
     scalacOptions ++= Seq(
-      "-Xfatal-warnings",
+//      "-Xfatal-warnings",
```
Comment (cavster) [inline build.sbt]: Why comment this out?

File: project/AppDependencies.scala [hunk 5]
```diff
@@ -4,28 +4,30 @@ import sbt._
...
+    "uk.gov.hmrc" %% "time-to-pay-taxpayer-cor" % "0.40.0",
 
-    "uk.gov.hmrc" %% "domain" %  "5.9.0-play-26",
-    "com.typesafe.play" %% "play-json-joda" % "2.6.13",
+    "uk.gov.hmrc" %% "domain" %  "6.0.0-play-28",
+    "com.typesafe.play" %% "play-json-joda" % "2.7.4",
 
-    "com.beachape" %% "enumeratum-play" % "1.5.15", // Nb: Staying below version 1.5.16 whilst https://github.com/lloydmeta/enumeratum/issues/285 causes play-2.7 to be pulled in
-    "org.julienrf" %% "play-json-derived-codecs" % "4.0.1"// Nb: Choose carefully for version of play. 4.0.1 is latest for play 2.6.
+    "com.beachape" %% "enumeratum-play" % "1.6.1", // Nb: Staying below version 1.5.16 whilst https://github.com/lloydmeta/enumeratum/issues/285 causes play-2.7 to be pulled in
```
Comment (cavster) [inline project/AppDependencies.scala]: If we are at play 2.8 we can remove these comments https://github.com/lloydmeta/enumeratum/issues/285 causes play-2.7 to be pulled in

File: app/testonly/TestUsersController.scala (hunk 3 above)
Comment (PucilowskiHMRC) [reply app/testonly/TestUsersController.scala:62]: L100 doesn't exist in the latest release of hmrc's auth-client (I think confidence levels are generally on the way out?)

this controller sets up test users for manual testing so won't affect business logic

**Review by cavster — APPROVED** [review]

File: app/ssttpcalculator/CalculatorController.scala (hunk 1 above)
Comment (cavster) [reply app/ssttpcalculator/CalculatorController.scala:96]: Ok we shall have to give it some Love

File: app/testonly/TestUsersController.scala (hunk 3 above)
Comment (cavster) [reply app/testonly/TestUsersController.scala:62]: ok

File: app/ssttpcalculator/CalculatorService.scala (hunk 2 above)
Comment (PucilowskiHMRC) [reply app/ssttpcalculator/CalculatorService.scala:235]: tried to rewrite it as tail-recursive but didn't pass tests, will leave as is seeing as it's been serving in production well

File: build.sbt (hunk 4 above)
Comment (PucilowskiHMRC) [reply build.sbt]: good spot, thanks!
